File: app/src/main/java/com/digestit/domain/model/Episode.kt
package com.digestit.domain.model

import java.time.Instant

enum class Platform(val displayName: String) {
    BILIBILI("哔哩哔哩"),
    XIAOYUZHOU("小宇宙"),
    UNKNOWN("未知平台")
}

enum class ProcessingStatus {
    QUEUED,
    EXTRACTING,
    TRANSCRIBING,
    SUMMARIZING,
    COMPLETED,
    FAILED
}

data class Episode(
    val id: String,
    val platform: Platform,
    val originalUrl: String,
    val title: String,
    val author: String,
    val coverUrl: String?,
    val durationSeconds: Int,
    val createdAt: Instant,
    val processingStatus: ProcessingStatus,
    val errorMessage: String? = null,
    val isFavorite: Boolean = false,
    val lastOpenedAt: Instant? = null,
    val audioUrl: String? = null,
)

data class ProcessingJob(
    val jobId: String,
    val episodeId: String?,
    val status: ProcessingStatus,
    val progress: Float,
    val currentStep: String,
    val errorMessage: String?
)

data class Transcript(
    val episodeId: String,
    val fullText: String,
    val segments: List<TranscriptSegment>,
    val language: String,
    val wordCount: Int
)

data class TranscriptSegment(
    val startMs: Long,
    val endMs: Long,
    val text: String
)

data class Summary(
    val episodeId: String,
    val oneLiner: String,
    val keyPoints: List<String>,
    val topics: List<String>,
    val highlights: List<Highlight>,
    val fullSummary: String
)

data class Highlight(
    val quote: String,
    val timestampMs: Long,
    val context: String
)

data class ChatSession(
    val id: String,
    val episodeId: String,
    val createdAt: Instant,
    val messages: List<ChatMessage>
)

data class ChatSessionInfo(
    val sessionId: String,
    val createdAt: Instant,
    val messageCount: Int,
    val previewText: String
)

data class ChatMessage(
    val id: String,
    val sessionId: String,
    val role: MessageRole,
    val content: String,
    val timestamp: Instant,
    val isStreaming: Boolean = false,
    val referencedTimestamps: List<Long> = emptyList()
)

enum class MessageRole { USER, ASSISTANT }

enum class SearchSourceType {
    TITLE,
    SUMMARY,
    TRANSCRIPT
}

data class SearchResult(
    val episodeId: String,
    val title: String,
    val subtitle: String,
    val previewText: String,
    val sourceType: SearchSourceType,
    val timestampMs: Long? = null,
)

data class BackendComponentHealth(
    val status: String,
    val detail: String? = null,
)

data class BackendWhisperHealth(
    val status: String,
    val mode: String,
    val model: String,
    val configuredMode: String,
    val initialized: Boolean,
    val fallback: Boolean,
    val errorMessage: String? = null,
)

data class BackendHealth(
    val status: String,
    val api: BackendComponentHealth,
    val db: BackendComponentHealth,
    val redis: BackendComponentHealth,
    val celery: BackendComponentHealth,
    val whisper: BackendWhisperHealth,
)

File: backend/app/models/episode.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import Column, String, Integer, DateTime, Text, Float, Enum as SAEnum
from sqlalchemy.orm import declarative_base
import enum

Base = declarative_base()


class Platform(str, enum.Enum):
    BILIBILI = "BILIBILI"
    XIAOYUZHOU = "XIAOYUZHOU"
    UNKNOWN = "UNKNOWN"


class ProcessingStatus(str, enum.Enum):
    QUEUED = "QUEUED"
    EXTRACTING = "EXTRACTING"
    TRANSCRIBING = "TRANSCRIBING"
    SUMMARIZING = "SUMMARIZING"
    COMPLETED = "COMPLETED"
    FAILED = "FAILED"


class Episode(Base):
    __tablename__ = "episodes"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    platform = Column(SAEnum(Platform), nullable=False)
    original_url = Column(String, nullable=False)
    title = Column(String, default="")
    author = Column(String, default="")
    cover_url = Column(String, nullable=True)
    duration_seconds = Column(Integer, default=0)
    audio_file_path = Column(String, nullable=True)
    created_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))
    processing_status = Column(SAEnum(ProcessingStatus), default=ProcessingStatus.QUEUED)
    error_message = Column(Text, nullable=True)


class ProcessingJob(Base):
    __tablename__ = "processing_jobs"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    episode_id = Column(String, nullable=True)
    celery_task_id = Column(String, nullable=True)
    status = Column(SAEnum(ProcessingStatus), default=ProcessingStatus.QUEUED)
    progress = Column(Float, default=0.0)
    current_step = Column(String, default="等待中")
    error_message = Column(Text, nullable=True)
    created_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))


class Transcript(Base):
    __tablename__ = "transcripts"

    episode_id = Column(String, primary_key=True)
    full_text = Column(Text, nullable=False)
    segments_json = Column(Text, nullable=False, default="[]")
    language = Column(String, default="zh")
    word_count = Column(Integer, default=0)


class Summary(Base):
    __tablename__ = "summaries"

    episode_id = Column(String, primary_key=True)
    one_liner = Column(Text, default="")
    key_points_json = Column(Text, default="[]")
    topics_json = Column(Text, default="[]")
    highlights_json = Column(Text, default="[]")
    full_summary = Column(Text, default="")


class ChatMessage(Base):
    __tablename__ = "chat_messages"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    session_id = Column(String, nullable=False, index=True)
    episode_id = Column(String, nullable=False, index=True)
    role = Column(String, nullable=False)  # "user" or "assistant"
    content = Column(Text, nullable=False)
    created_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))
